Software/real_time/ROS_RoboBuggy/src/robobuggy/scripts/data_map_plot.py

```diff
@@ -1,20 +1,82 @@
 #!/usr/bin/env python
 
 import rospy
+import math
+import json
+import time
+import os
+from gps_common.msg import GPSFix
 from robobuggy.msg import GPS
+from robobuggy.msg import Pose
+from robobuggy.msg import Command
+
+viz_pub = None;
+
+def pose_callback(data):
+    global viz_pub
+
+    rospy.loginfo("got Pose msg: %f degrees lat, %f degrees long, %f rad", data.latitude_deg, data.longitude_deg, data.heading_rad)
+
+    #data.heading_rad is in Radians from north clockwise, but GPSFix requires degrees from north
+    degrees_from_north = (180 * data.heading_rad) / math.pi
+
+    last_latitude = data.latitude_deg
+    last_longitude = data.longitude_deg
+    last_heading = degrees_from_north
+    
+    viz_msg_heading = GPSFix(latitude=data.latitude_deg, longitude=data.longitude_deg, track = degrees_from_north)
+ 
+    viz_pub.publish(viz_msg_heading)
 
-def subscriber_callback(data):
-    rospy.loginfo("got gps msg: %f, %f", data.Lat_deg, data.Lon_deg)
+    pass
+
+def command_callback(data):
+    global viz_pub
+    
+    #steer_cmd should be degrees
+    rospy.loginfo("got Steering Command msg: %f degrees", data.steer_cmd)
+
+    viz_msg_steering = GPSFix(latitude=last_latitude, longitude=last_longitude, track = last_heading + data.steer_cmd)
+ 
+    viz_pub.publish(viz_msg_steering)
+
+    pass
+
+
+def waypoints_publisher():
+    waypoints_pub = rospy.Publisher('WAYPOINTS_VIZ',GPSFix, queue_size=10)
+    
+    # read from waypoints file, parse JSON, publish
+    waypoints = open("../RoboBuggy/Software/real_time/ROS_RoboBuggy/src/robobuggy/config/waypoints.txt", 'r')
```

Comment:
Not something that needs to get done now, but consider making this a parameter from the launch file